packages/core/contracts/financial-templates/common/financial-product-libraries/PreExpirationIdentifierTransformationFinancialProductLibrary.sol

```diff
@@ -0,0 +1,59 @@
+pragma solidity ^0.6.0;
+pragma experimental ABIEncoderV2;
+import "./FinancialProductLibrary.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+/**
+ * @title Pre-Expiration Identifier Transformation Financial Product Library
+ * @notice Adds custom identifier transformation to enable a financial contract to use two different identifiers, depending
+ * on when a price request is made. If the request is made before expiration than a transformation is made to the identifier
+ * & if it is at or after expiration then the original identifier is returned. This library enables self referential
+ * TWAP identifier to be used on synthetics pre-expiration, in conjunction with a separate identifier at expiration.
+ */
+contract PreExpirationIdentifierTransformationFinancialProductLibrary is FinancialProductLibrary, Ownable {
+    mapping(address => bytes32) financialProductTransformedIdentifiers;
+
+    /**
+     * @notice Enables the deployer of the library to set the transformed identifier for an associated financial product.
+     * @param financialProduct address of the financial product.
+     * @param transformedIdentifier the identifier for the financial product to be used if the contract is post expiration.
+     * @dev Note: a) Only the owner (deployer) of this library can set identifier transformations b) The identifier Cant
+     * be set to blank. c) A transformed price can only be set once to prevent the deployer from changing it after the fact.
+     * d)  financialProduct must exposes an expirationTimestamp method.
+     */
+    function setFinancialProductTransformedIdentifier(address financialProduct, bytes32 transformedIdentifier)
+        public
+        onlyOwner
+    {
+        require(transformedIdentifier != "", "Cant set to empty transformation");
+        require(financialProductTransformedIdentifiers[financialProduct] == "", "Transformation already set");
+        require(ExpiringContractInterface(financialProduct).expirationTimestamp() != 0, "Invalid EMP contract");
+        financialProductTransformedIdentifiers[financialProduct] = transformedIdentifier;
```

Comment:
We could, but IIUC, the transaction will fail later for the same reason, so this doesn't really change the outcome. One advantage of _not_ checking for this is that this contract doesn't require the finder or any other DVM contracts to work.

Reply:
I agree with @mrice32. Also, this library is really just meant as an example for now and I suspect it might require small changes potentially for a real product using this kind of transformation logic so better to keep it as simple as possible for now.